src/adapters/scene.js

```diff
@@ -26,30 +26,39 @@ const LONG_TOUCH_DELAY_MS = 500;
 function Scene() {
   this.currentMarker = null;
   this.popup = new PoiPopup();
-  this.zoom = map.zoom;
-  this.center = [map.center.lng, map.center.lat];
   this.savedLocation = null;
 }
 
-Scene.prototype.initScene = async function(locationHash) {
-  await this.setupInitialPosition(locationHash);
-  this.initMapBox();
-};
-
-Scene.prototype.setupInitialPosition = async function(locationHash) {
+Scene.prototype.getMapInitOptions = async function(locationHash) {
   if (locationHash) {
-    this.zoom = locationHash.zoom;
-    this.center = [locationHash.lng, locationHash.lat];
-  } else {
-    const lastLocation = await store.getLastLocation();
-    if (lastLocation) {
-      this.center = [lastLocation.lng, lastLocation.lat];
-      this.zoom = lastLocation.zoom;
-    }
+    return {
+      zoom: locationHash.zoom,
+      center: [locationHash.lng, locationHash.lat],
+    };
   }
+  const lastLocation = await store.getLastLocation();
+  if (lastLocation) {
+    return {
+      zoom: lastLocation.zoom,
+      center: [lastLocation.lng, lastLocation.lat],
+    };
+  }
+  if (window.initialBbox) {
+    return {
+      bounds: window.initialBbox,
+      fitBoundsOptions: {
+        padding: this.getCurrentPaddings(),
+        maxZoom: 9,
+      },
+    };
+  }
+  return {
+    zoom: map.zoom,
```

Comment:
Cool simplification of the map init code, it's simpler to follow this way. The only thing I had to trace back in the code was this `map` constant, which is ambiguously named in this context. What do you think about renaming it to `mapConfig` on import while we are changing this code?

Reply:
Sure, it's done.